api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -136,37 +136,53 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
+	allErrs = append(allErrs, m.ValidateCommonFields(specPath)...)
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(GroupVersion.WithKind("MachineHealthCheck").GroupKind(), m.Name, allErrs)
+}
+
+// ValidateCommonFields validates UnhealthyConditions NodeStartupTimeout, MaxUnhealthy, and RemediationTemplate of the MHC.
+// It is also used to help validate other types which define MachineHealthChecks such as MachineHealthCheckClass and MachineHealthCheckTopology.
+// Note: this function is used for internal validation and is not intended for external consumption.
+func (m *MachineHealthCheck) ValidateCommonFields(fldPath *field.Path) field.ErrorList {
```

Comment:
only one nit about naming because Common is relly generic
what about ValidateTopologyFields?

Also, the note is kind of confusing. if this is a public method part of the API it is hard to say not for external consumption... if we want this to be an internal method let's move it under internal/topology(/someting)

Reply:
To be honest with this implementation I would prefer Common because Topology doesn't sound right (does fields are not primarily fields of the Topology, they are shared between MHC and the MHC in ClusterClass and Cluster.spec.topology).

> Also, the note is kind of confusing. if this is a public method part of the API it is hard to say not for external consumption... if we want this to be an internal method let's move it under internal/topology(/someting)

Good point, but I wouldn't move it into a new package.

I played around a bit and we can just move the func into `internal/webhooks`

I open a PR on top of the PR to showcase how it woud look like

Reply:
Okay just realized that doesn't work because of a cyclic dependency.

So tl;dr we can't move this func into another package if we use `MachineHealthCheck` in the func signature.
api/v1beta1 => internal/webhook (to call the func)
internal/webhooks => api/v1beta1 (for the API type)

The only way to move this code out of the api package is as far as I can tell to move the entire webhook code like we did for Cluster & ClusterClass. But not sure if that's what we want to do.

The idea behind the disclaimer was just to give folks a warning that they shouldn't rely on this code (Just the same as they shouldn't really rely on ValidateCreate, ValidateUpdate, ValidateDelete on our API types).

But we can also drop the disclaimer and then we just deprecate the func if we ever want to get rid of it.

Reply:
IMO we should drop the disclaimer as it doesn't really work to create a public API but state it shouldn't be used. I think adding this function here is fine though - there's no need to really add it to an internal package given ValidateCreate etc. are exposed here.

Alternative is to make a webhook-only func which copies this one and link them with a comment / ref to ensure they're updated in line (this func probably won't be updated much)